news/10018-add-environment-variable-expansion-for-repository-url

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* add environment variable expansion for repository url (#10018)
```

Comment:
It might be worth elaborating in more detail how this looks like, as it IS a change of behavior and might be useful for end-users to read in the changelog about